SC_redesign/Crop_and_Soil/soil/evapotranspiration.py

```diff
@@ -0,0 +1,264 @@
+from math import exp
+from typing import Optional
+
+from SC_redesign.Crop_and_Soil.soil.layer_data import LayerData
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+
+
+class Evapotranspiration:
+    def __init__(self, soil_data: Optional[SoilData] = None):
+        self.data = soil_data or SoilData()  # initialize with defaults, if not given
+
+    # --- main routine ---
+    def evapotranspirate(self, extraterrestrial_radiation: float, max_air_temp: float, min_air_temp: float,
+                         avg_air_temp: float, above_ground_biomass: float, residue: float, snow_water_content: float,
+                         initial_canopy_free_water: float) \
+            -> None:
+        """does the evapotranspiration of the soil on a given day
+
+        Details: calculates and stores the potential evapotranspiration, soil evaporation in the SoilData object
+        """
+        self.data.potential_evapotranspiration = self._determine_potential_evapotranspiration(
+            extraterrestrial_radiation,
+            max_air_temp,
+            min_air_temp,
+            avg_air_temp)
+        self.data.potential_evapotranspiration_adjusted = self._determine_potential_evapotranspiration_adjusted(
+            initial_canopy_free_water)
+        # TODO: add field (in CropData?) to track amount of free water in canopy as it gets adjusted - issue #316
```

Comment:
```suggestion
        # TODO: add attribute (in CropData?) to track amount of free water in canopy as it gets adjusted - issue #316
```

Let's avoid using the term "field" to refer to anything other than an agricultural field, if possible.